app/es_api/utils/queries.py

```diff
@@ -211,6 +213,45 @@ def more_like_this(self, doc_id):
 
         return response
 
+    def similar_courses(self, keyword=""):
+        """This method takes in a doc ID and queries the elasticsearch index for
+            courses with similar competencies or titles"""
+
+        course_mapping = CourseInformationMapping.objects.first()
+        fields = [
+            course_mapping.course_competency,
+            course_mapping.course_subject
+        ]
+
+        # We're going to match based only on two fields
+        q = Q("multi_match",
+              query=keyword,
+              fields=fields)
+
+        # setting up the search object
+        self.search = self.search.query(q)
+
+        self.user_organization_filtering()
+
+        # getting the page size for result pagination
+        configuration = XDSConfiguration.objects.first()
+        uiConfig = configuration.xdsuiconfiguration
+        search_filters = SearchFilter.objects.filter(
+            xds_ui_configuration=uiConfig, active=True)
+
+        # create aggregations for each filter
+        self.add_search_aggregations(filter_set=search_filters)
+
+        # Skipping first response found and getting next three
+        # if len(self.search > 3):
+        self.search = self.search[1:4]
```

Comment:
@mdonlon why are you skipping the first? also probably don't need filters or aggregation

Reply:
you should probably keep the user_organization_filtering in.  
what is the value of keyword coming in?  
if it is a UUID then you can either add that to the search query as not equal, or get 4 items returned and if one matches the UUID remove it from the list.  if keyword is a UUID you would also need to update your query.
if it is a competency or subject, then you could consider including the UUID of the object you are basing it off of as a parameter so you can add it to the search, or just send 4 responses to the UI and let the UI filter out any.

Reply:
I'm sending what is being mapped into the competency field (currently KSA Alignment for DOT&E courses). I think the easiest way to is to send back 4 and have the UI filter out one if the UUID matches the UUID of what we used to send out the query in the first place. This query also might not be getting used in our Portal code, or having the same course might not be as big of an issue going forward.